build/ci/send-to-helix.yml

```diff
@@ -22,6 +22,8 @@ parameters:
 steps:
   - script: ${{ parameters.MsBuildScript}}
             $(Build.SourcesDirectory)/eng/helix.proj
+            /t:Restore
```

Comment:
This doesn't work since it's a single evaluation.  You need to either do two calls to msbuild, first restore, then build -- or you can use dotnet build which will itself call restore before building.